app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt

```diff
@@ -93,24 +93,29 @@ class SearchFragment : Fragment() {
         menu?.setGroupVisible(R.id.search_menu, true)
         menu?.setGroupVisible(R.id.profile_menu, false)
 
-        val searchView:SearchView ?= menu?.findItem(R.id.search_item)?.actionView as? SearchView
-        searchView?.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+        val searchView: SearchView? = menu?.findItem(R.id.search_item)?.actionView as? SearchView
+        val queryListener = object : SearchView.OnQueryTextListener {
             override fun onQueryTextSubmit(query: String): Boolean {
                 //Do your search
                 searchViewModel.searchEvent = query
                 rootView.search_linear_layout.visibility = View.GONE
-                if (searchViewModel.savedLocation != null && TextUtils.isEmpty(rootView.location_editText.text.toString()))
+                rootView.fabSearch.visibility = View.GONE
+                if (searchViewModel.savedLocation != null && TextUtils.isEmpty(rootView.locationEditText.text.toString()))
                     searchViewModel.loadEvents(searchViewModel.savedLocation.nullToEmpty())
                 else
-                    searchViewModel.loadEvents(rootView.location_editText.text.toString().nullToEmpty())
+                    searchViewModel.loadEvents(rootView.locationEditText.text.toString().nullToEmpty())
                 loadEventsAgain = true
                 return false
             }
 
             override fun onQueryTextChange(newText: String): Boolean {
                 return false
             }
-        })
+        }
+        searchView?.setOnQueryTextListener(queryListener)
```

Comment:
Don't you think, you should set it back to null in the `onDestroy` or equivalent ?

Reply:
to prevent memory leak? yes i guess

Reply:
@iamareebjamal Will it cause a memory leak?

Reply:
Yes